qubes-gui-gntalloc/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+//! Grant-table manipulation code
+
+use std::io;
+use std::os::unix::io::AsRawFd as _;
+use std::rc::{Rc, Weak};
+
+type DomID = u16;
+
+/// A GUI agent instance
+pub struct Agent {
+    alloc: Rc<std::fs::File>,
+    peer: DomID,
+}
+
+/// A window sent to the GUI daemon
+pub struct Buffer {
+    inner: Vec<u64>,
```

Comment:
`inner` is a very non-descriptive name for a struct that also has four other fields. I'd say doc-comments are warranted for most of the fields here. In particular, one fact worth mentioning is that this vector logically contains `WindowDumpHeader` followed by `u32`s, and the `u64` is there only to get the right alignment.